charts/shoot-core/components/charts/coredns/templates/coredns-networkpolicy.yaml

```diff
@@ -0,0 +1,31 @@
+apiVersion: {{ include "networkpolicyversion" . }}
+kind: NetworkPolicy
+metadata:
+  name: gardener.cloud--allow-dns
+  namespace: {{ .Release.Namespace }}
+spec:
+  podSelector:
+    matchExpressions:
+    - {key: k8s-app, operator: In, values: [kube-dns,coredns]}
+  egress:
+  - ports:
+    - port: 443
+      protocol: TCP
+    - port: 53
```

Comment:
The `443` port is so that the core dns pods can talk to the API server I suppose?
The `53` ports are so that when  the node local DNS deamon set pods make requests to coredns (due to cache miss), egress from coredns is allowed back to the node local DNS deamon set pods?

Would be helpful to have comments in the code.